### PR TITLE
[CUMULUS-3033] Fix granuleEsQuery infinite loop bug 

### PR DESCRIPTION
**Pull request by botanical**
**Summary:** Summary of changes

Addresses [CUMULUS-3033:Bulk operation elastic search specified with wrong time period will cause the async task run forever](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3033)

## Changes

* Fix granuleEsQuery infinite loop bug in the case that wrong time period is provided and `hits.total.value` is 0.

## Testing details
I tested by deploying my changes to our SIT account and running a bulk operation for granules where the queries contain a timestamp range that would cause the operation to run forever:
```
{
  "workflowName": "BulkOperationsWorkflowTest",
  "index": "cumulus-std-granule-*",
  "query": {
    "query": {
      "bool": {
        "must": [],
        "filter": [
          {
            "range": {
              "@timestamp": {
                "gte": "2022-11-07T23:59:00.220Z",
                "lte": "2022-11-08T19:51:36.220Z",
                "format": "strict_date_optional_time"
              }
            }
          }
        ],
        "should": [],
        "must_not": []
      }
    }
  },
  "ids": [],
  "meta": {}
}
```

and with timeout specified
```
{
  "workflowName": "BulkOperationsWorkflowTest",
  "index": "cumulus-std-granule-*",
  "query": {
    "query": {
      "bool": {
        "must": [],
        "filter": [
          {
            "range": {
              "@timestamp": {
                "gte": "2022-11-07T23:59:00.220Z",
                "lte": "2022-11-08T19:51:36.220Z",
                "format": "strict_date_optional_time"
              }
            }
          }
        ],
        "should": [],
        "must_not": []
      }
    }
  },
  "ids": [],
  "meta": {}
}
```
Instead of the status of the operation being stuck with `Running` it will now appropriately succeed.
## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

**Comment by jennyhliu** [comment]
Should the query with endtime < starttime return 0 granule?  And then the bulk operation just finishes successfully without acting on any granule.

**Comment by botanical** [comment]
@Nnaga1 @jennyhliu I spoke too soon, need to continue working on this since I was testing with the wrong index and the "stuck in running" problem is still an issue after using the correct index for testing 🤦‍♀️

**Review by jennyhliu — CHANGES_REQUESTED** [review]
bulk operation should succeed even query doesn't return any granule

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
   - Added `@cumulus/api/endpoints/granules` unit to cover duration overwrite
     logic for PUT/PATCH endpoint.
+- **CUMULUS-3033**
+  - Added default timeout for `granuleEsQuery` search operation to be 60 seconds.
+  - Added functionality to allow search operation timeout to be configurable.
+
+### Fixed
+
+- **CUMULUS-3033**
+  - Fixed `granuleEsQuery` to properly terminate if `body.hit.total.value` is 0.
```
Comment (jennyhliu) [inline CHANGELOG.md:46]: Is timeout still an issue after you make this change?  If not, we should remove the `timeout` update.

Comment (botanical) [reply CHANGELOG.md:43]: https://github.com/nasa/cumulus/pull/3201/commits/c74d76ab0f8d7a18f77c21f1dc63954fdada4258

**Comment by jennyhliu** [comment]
please update PR title and descriptions.

File: packages/api/tests/lib/test-granules.js [hunk 2]
```diff
@@ -344,3 +348,62 @@ test('translateGranule() will translate an old-style granule file and numeric pr
...
+    },
+    max_score: null,
+    hits: [],
+  };
+
+  t.truthy(await granuleEsQuery({
+    index: esIndex,
+    query,
+    source: ['granuleId'],
+    testBodyHits,
```
[the remark this answers is not in the document]
Comment (botanical) [reply packages/api/tests/lib/test-granules.js:407]: No, the query is returning `hit.total` as 0, not `hits.total.value`

[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/tests/lib/test-granules.js:407]: When does the query returns `hits.total.value` then?

[the remark this answers is not in the document]
Comment (botanical) [reply packages/api/tests/lib/test-granules.js:407]: The ad hoc testing and unit testing does not yield the same search results given the same query. When you deploy to SIT and query, then `hits.total.value` is 0 given the query above.

[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/tests/lib/test-granules.js:407]: I'm wondering if that's caused by the different es versions (cloud metrics vs cumulus)

[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/tests/lib/test-granules.js:407]: Cloud metrics does return `hits.total.value`:
{"took":2,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":10000,"relation":"gte"},"max_score":1.0,"hits":[{"_index":"...

[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/tests/lib/test-granules.js:407]: Suggest to make a comment on the code to state that some es like Cloud Metrics returns `hits.total.value`

[the remark this answers is not in the document]
Comment (botanical) [reply packages/api/tests/lib/test-granules.js:407]: https://github.com/nasa/cumulus/pull/3201/commits/a51ed4da7d997721e27a091ad0976001154a782c

**Review by jennyhliu — APPROVED** [review]
Nice work!

**Pull request merged**